### PR TITLE
Add "memory" functionality

### PR DESCRIPTION
**Pull request by ramadantu**

### CODE REVIEW

File: src/main/java/com/internship/project/ProjectApplication.java [hunk 1]
```diff
@@ -1,47 +1,102 @@
...
-        final String VAR_VALUE = "CALC_VAR_VALUE";
+        try (InputStream inputStream = ProjectApplication.class.getResourceAsStream("/calculator.properties");
+             FileReader fileReader = new FileReader("src/main/resources/input.txt");
+             FileWriter fileWriter = new FileWriter("src/main/resources/memory.txt");
+             BufferedReader bufferedReader = new BufferedReader(fileReader)) {
 
-        try (InputStream inputStream = ClassLoader.getSystemResourceAsStream("calculator.properties");
-             Scanner scanner = new Scanner(System.in)) {
+            DatabaseH2 databaseH2 = new DatabaseH2();
+            databaseH2.deleteTableIfExists();
```
Comment (ghost) [inline src/main/java/com/internship/project/ProjectApplication.java]: Why clear the table in an in-memory DB? 🤔

File: src/main/java/com/internship/project/ProjectApplication.java [hunk 2]
```diff
@@ -1,47 +1,102 @@
...
-                properties.setProperty(VAR_NAME, envVarName);
-                properties.setProperty(VAR_VALUE, envVarValue);
+            while ((inputLine = bufferedReader.readLine()) != null) {
+
+                leftSideOfExpression = inputLine.substring(0, inputLine.indexOf("="));
+                rightSideOfExpression = inputLine.substring(inputLine.indexOf("="));
 
-                String variableName = properties.getProperty(VAR_NAME);
-                String variableValue = properties.getProperty(VAR_VALUE);
+                if (envVarName != null && envVarValue != null) {
```
Comment (ghost) [inline src/main/java/com/internship/project/ProjectApplication.java]: Possibility to use Optionals https://www.baeldung.com/java-optional

File: src/main/java/com/internship/project/ProjectApplication.java [hunk 3]
```diff
@@ -1,47 +1,102 @@
...
+            databaseH2.insertRecord(properties.getProperty(MEM_VALUE));
+            return rightSideOfExpression.replace("M-", properties.getProperty(MEM_VALUE));
+        } else if (rightSideOfExpression.contains("M")) {
+            properties.setProperty(MEM_VALUE, resultOfLeftSide);
+            fileWriter.append(String.format(MEM_FILE_FORMAT, properties.getProperty(MEM_VALUE)));
+            databaseH2.insertRecord(properties.getProperty(MEM_VALUE));
+            return rightSideOfExpression.replace("M", properties.getProperty(MEM_VALUE));
+        } else {
+            return rightSideOfExpression.replace("?", resultOfLeftSide);
+        }
```
Comment (ghost) [inline src/main/java/com/internship/project/ProjectApplication.java]: Possibly extract functionallity into class.

File: src/main/java/com/internship/project/database/DatabaseH2.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
+package com.internship.project.database;
+
+import java.sql.*;
+
+public class DatabaseH2 {
+    public final String JDBC_DRIVER = "org.h2.Driver";
+    public final String DB_URL = "jdbc:h2:~/test";
+    public final String USER = "ramadan";
+    public final String PASS = "123456";
```
Comment (ghost) [inline src/main/java/com/internship/project/database/DatabaseH2.java]: Possibly extract into a .properties file

File: src/main/java/com/internship/project/database/DatabaseH2.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                    stmt.close();
+                }
+                if (conn != null) {
+                    conn.close();
+                }
+            } catch (SQLException sqlE) {
+                sqlE.printStackTrace();
+            }
+        }
+    }
```
Comment (vap78) [inline src/main/java/com/internship/project/database/DatabaseH2.java]: Split this into two methods - updateDB and queryDB.
Even better specify the exact operation in the method name like:

updateCalculatorMemory()
fetchCalculatorMemory()

File: src/main/java/com/internship/project/database/DatabaseH2.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        databaseUpdate("CREATE TABLE   Calculator_Memory " +
+                "(id INTEGER not NULL, " +
+                " M DOUBLE, " +
+                " PRIMARY KEY ( id ))");
+    }
+
+    public void insertRecord(String value) {
+        databaseUpdate("INSERT INTO Calculator_Memory " +
+                "VALUES(" + ID++ + ", " + value + ")");
+    }
```
Comment (ghost) [inline src/main/java/com/internship/project/database/DatabaseH2.java]: https://www.baeldung.com/java-statement-preparedstatement

**Review by ghost — COMMENTED** [review]
Good Job 🤵‍♂️

**Pull request merged**